src/qibolab/instruments/simulator/backends/qutip_backend.py

```diff
@@ -0,0 +1,588 @@
+"""
+qutip_backend.py
+----------------
+This module provides a backend for Quantum Toolbox in Python (QuTiP) to simulate QPUs.
+"""
+
+from collections import OrderedDict
+from timeit import default_timer as timer
+from typing import List, Optional
+
+import numpy as np
+from qutip import Options, Qobj, basis, expect, ket2dm, mesolve, ptrace
+from qutip.operators import identity as Id
+from qutip.tensor import tensor
+from qutip.ui.progressbar import EnhancedTextProgressBar
+
+from qibolab.instruments.simulator.backends.generic import (
+    dec_to_basis_string,
+    op_from_instruction,
+)
+
+
+def get_default_qutip_sim_opts():
+    """Returns the default simulation options for the Qutip backend.
+
+    Returns:
+        Options: The default simulation options.
+    """
+    sim_opts = Options(atol=1e-11, rtol=1e-9, nsteps=int(1e6))
+    sim_opts.normalize_output = False  # mesolve is x3 faster if this is False
+
+    return sim_opts
+
+
+class Qutip_Simulator:
```

Comment:
Please use usual name conventions: https://peps.python.org/pep-0008/#class-names
```suggestion
class QutipSimulator:
```
This may be relevant for other places in the code too.

Reply:
Fixed in 539becaa7b31e32a04a2d8e32e7a92c412a681b1